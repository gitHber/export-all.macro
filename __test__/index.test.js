const pluginTester = require("babel-plugin-tester");
const path = require("path");
const plugin = require("babel-plugin-macros");

pluginTester.default({
  pluginName: "export",
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename, parserOpts: {} },
  tests: {
    "default export": `
      import exportAll from "../macro";
      exportAll("./", {exclude: ["**/*.test.js"]});
    `,
    "export all": `
      import exportAll from "../macro";
      exportAll("./", {exclude: ["**/*.test.js"], onlyIndex: false});
    `,
    "export noDefault": `
      import exportAll from "../macro";
      exportAll("./", {exclude: ["**/*.test.js"], noDefault: true});
    `,
  },
});
