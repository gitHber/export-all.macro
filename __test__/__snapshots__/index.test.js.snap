// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export default export: default export 1`] = `

import exportAll from "../macro";
exportAll("./", {exclude: ["**/*.test.js"]});

      ↓ ↓ ↓ ↓ ↓ ↓

export * from "./Button.tsx";
export { default as Btn } from "./Btn/index.js";
export * from "./Btn/index.js";


`;

exports[`export export all: export all 1`] = `

import exportAll from "../macro";
exportAll("./", {exclude: ["**/*.test.js"], onlyIndex: false});

      ↓ ↓ ↓ ↓ ↓ ↓

export * from "./Button.tsx";
export { default as Btn } from "./Btn/index.js";
export * from "./Btn/index.js";
export { default as button } from "./Btn/button.js";
export * from "./Btn/button.js";


`;

exports[`export export noDefault: export noDefault 1`] = `

import exportAll from "../macro";
exportAll("./", {exclude: ["**/*.test.js"], noDefault: true});

      ↓ ↓ ↓ ↓ ↓ ↓

export * from "./Button.tsx";
export * from "./Btn/index.js";


`;
